paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/RepositoryLoader.kt

```diff
@@ -0,0 +1,919 @@
+import com.android.ide.common.util.toPathString
+import com.android.io.CancellableFileIo
+import com.android.resources.Arity
+import com.android.resources.Density
+import com.android.resources.FolderTypeRelationship
+import com.android.resources.ResourceFolderType
+import com.android.resources.ResourceFolderType.VALUES
+import com.android.resources.ResourceType
+import com.android.resources.ResourceType.ANIMATOR
+import com.android.resources.ResourceType.ARRAY
+import com.android.resources.ResourceType.ATTR
+import com.android.resources.ResourceType.DRAWABLE
+import com.android.resources.ResourceType.ID
+import com.android.resources.ResourceType.INTERPOLATOR
+import com.android.resources.ResourceType.LAYOUT
+import com.android.resources.ResourceType.MENU
+import com.android.resources.ResourceType.MIPMAP
+import com.android.resources.ResourceType.PLURALS
+import com.android.resources.ResourceType.PUBLIC
+import com.android.resources.ResourceType.STRING
+import com.android.resources.ResourceType.STYLE
+import com.android.resources.ResourceType.STYLEABLE
+import com.android.resources.ResourceType.TRANSITION
+import com.android.resources.ResourceVisibility
+import com.android.resources.ResourceVisibility.PRIVATE
+import com.android.utils.SdkUtils
+import com.android.utils.XmlUtils
+import com.google.common.collect.Maps
+import com.google.common.collect.Table
+import com.google.common.collect.Tables
+import org.kxml2.io.KXmlParser
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import java.io.BufferedInputStream
+import java.io.File
+import java.io.IOException
+import java.util.ArrayDeque
+import java.util.Deque
+import java.util.EnumMap
+import java.util.EnumSet
+
+/**
+ * Ported from: [RepositoryLoader.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/RepositoryLoader.java)
+ */
+abstract class RepositoryLoader<T : LoadableResourceRepository>(
+  private val namespace: ResourceNamespace
+) {
+
+  private val parser = ValueResourceXmlParser()
+  private val urlParser = ResourceUrlParser()
+  private val textExtractor: XmlTextExtractor = XmlTextExtractor()
+
+  private val valueFileResources: Table<ResourceType, String, BasicResourceItem> =
+    Tables.newCustomTable(
+      EnumMap(
+        ResourceType::class.java
+      )
+    ) { LinkedHashMap() }
+
+  protected fun loadResourceFile(
+    file: PathString,
+    configuration: RepositoryConfiguration
+  ) {
+    val folderType = getFolderType(file)
+    if (folderType == VALUES) {
+      if (isXmlFile(file)) {
+        parseValueResourceFile(file, configuration)
+      }
+    } else {
+      if (isXmlFile(file) && FolderTypeRelationship.isIdGeneratingFolderType(folderType)) {
+        parseIdGeneratingResourceFile(file, configuration)
+      }
+
+      parseFileResourceFile(file, folderType, configuration)
+    }
+  }
+
```

Comment:
@jrodbx Could you review the `loadResourceFile` parsing flow? Details are just copied from original,  just need to review overall flow to see anything confuse you? Thanks